frontend/models/Product.js

```diff
@@ -0,0 +1,23 @@
+const productsApiUrl = 'http://localhost:3000/products';
+
+class Product {
+    constructor(productData) {
+        this.id = productData.id,
+        this.name = productData.name,
+        this.description = productData.description,
+        this.price = productData.price,
+        this.currency = productData.currency,
+        this.imgUrl = productData.imgUrl,
+        this.comments = productData.comments
+    }
+
+    static getProductDetails(id) {
+        const productApiUrl = productsApiUrl + `/${id}`;
+        return fetch (productApiUrl, 

```

Comment:
Good job creating a separate class for Product, the getProductDetails to return a promise isn't a good practice. You can use async await it's a cleaner way or simply return the response when you're resolving the promise in the function.  
For example: 
instead of:  `Product.getProductDetails(id)
    .then(productData => renderProductDetails(productDetailsContainer,productData));`
you can use: `renderProductDetails(productDetailsContainer, Product.getProductDetails(id))`

If you create a class it's better if you do an `export default` and than `import`.